app/views/core/LoadingProgress.vue

```diff
@@ -53,14 +53,27 @@
             return 100
           }
 
+          const toPercent = val => {
+            if (_.isBoolean(val)) {
+              return val ? 0 : 100
+            } else if (!_.isNumber(val)) {
+              throw new Error('Percent must be array, boolean or percent')
+            }
+
+            return val
+          }
+
           const statuses = this.statuses.map((status) => {
-            if (typeof status === 'boolean') {
-              return status ? 0 : 100
-            } else if (typeof status !== 'number') {
-              throw new Error('Status must be boolean or percent')
+            if (!_.isArray(status)) {
+              return toPercent(status)
             }
 
-            return status
+            // The reduce function will sum up an initial boolean as 1,
+            // so we make sure it is following our concept of boolean percent instead.
+            status[0] = toPercent(status[0])
+
+            const loadingSum = status.reduce((total, toAdd) => total + toPercent(toAdd))
+            return loadingSum / status.length
```

Comment:
Similarly unless there is an invariant that the array is never empty, a division by 0 returns Infinity.